tests/puppeteer/pages/commonPage.js

```diff
@@ -10,4 +10,52 @@ module.exports = class CommonPage {
   async goTo(URL) {
     await this.page.goto(URL);
   }
+
+  /**
+   * Wait for selector and click
+   * @param selector, element to check
+   * @return boolean, true if text exist, false if not
+   */
+  async waitForSelectorAndClick(selector, timeout = 10) {
+    await this.page.waitForSelector(selector, {visible: true, timeout: timeout});
+    await this.page.click(selector)
+  }
+
+  /**
+   * Check text value
+   * @param selector, element to check
+   * @param textToCheckWith, text to check with
+   * @param parameter, parameter to use
+   * @return boolean, true when the result of parameter(textToCheckWith) is true, false if not
+   */
+  async checkTextValue(selector, textToCheckWith, parameter = 'equal') {
+    switch (parameter) {
+      case "equal":
+        await this.page.waitForSelector(selector);
+        await this.page.$eval(selector, el => el.innerText).then((text) => {
+          if (text.indexOf('\t') != -1) {
+            text = text.replace('\t', '');
+          }
```

Comment:
To test : maybe `el.textContent` will return text without char '\t' and no need to replace it and execute trim on it too

Reply:
@boubkerbribri  I tried this solution :
`await this.page.$eval(selector, el => el.innerText).then((text) => expect(text.replace(/\s+/g, ' ').trim()).to.equal(textToCheckWith));`
What do you think ?

Reply:
I think more this (but to check if textContent delete '\t')
```js
await this.page.$eval(selector, el => el.textContent)
.then((text) => expect(text.trim()).to.equal(textToCheckWith));
```